src/edu/washington/escience/myria/operator/SampledDbInsertTemp.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * 
+ */
+package edu.washington.escience.myria.operator;
+
+import java.io.File;
+import java.util.*;
+
+import com.almworks.sqlite4java.SQLiteConnection;
+import com.almworks.sqlite4java.SQLiteException;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.*;
+import edu.washington.escience.myria.accessmethod.AccessMethod;
+import edu.washington.escience.myria.accessmethod.ConnectionInfo;
+import edu.washington.escience.myria.accessmethod.SQLiteInfo;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.IntColumnBuilder;
+import edu.washington.escience.myria.parallel.RelationWriteMetadata;
+import edu.washington.escience.myria.storage.MutableTupleBuffer;
+import edu.washington.escience.myria.storage.TupleBatch;
+
+/**
+ * Samples the stream into a temp relation.
+ */
+public class SampledDbInsertTemp extends UnaryOperator implements DbWriter {
```

Comment:
Why couldn't this class extend `DbInsertTemp`? I think you should be able to override `consumeTuples()` to sample from each `TupleBatch` into the reservoir, and override `childEOS()` to insert the reservoir into the temp table.